packages/core/src/tools/sendToExtension.ts

```diff
@@ -0,0 +1,14 @@
+declare const __BUILD_ENV__SDK_VERSION__: string
+
+type ExtensionMessageType = 'logs' | 'record' | 'rum' | 'telemetry'
+
+export function sendToExtension(type: ExtensionMessageType, payload: unknown) {
+  // Only allow this for dev bundles for now, as it might impact performances?
+  if (__BUILD_ENV__SDK_VERSION__ === 'dev') {
+    window.dispatchEvent(
+      new CustomEvent('__ddBrowserSdkMessage', {
+        detail: { type, payload },
+      })
+    )
+  }
+}
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏IMO, for the performance impact alone, we could have an approach allowing us to use it with prod bundle as well by emitting sdk events only after a specific API has been called.
A bit like the debug mode with the `_setDebug` API.
However, that would open a new public api for others to integrate with us which could require a bit more thoughts on this API.
wdyt?

Reply:
Sure, let me try that! I want to make sure that whatever solution we chose, the very early events are captured (ex: those that are collected during `init`). That's why it could be a bit trickier to implement.